src/scripts/main.js

```diff
@@ -1,5 +1,26 @@
 'use strict';
 
+const wall = document.querySelector('.wall');
+const spider = wall.querySelector('.spider');
+
+const topMargin = wall.offsetTop + wall.clientTop + spider.clientHeight / 2;
+const leftMargin = wall.offsetLeft + wall.clientLeft + spider.clientWidth / 2;
```

Comment:
The calculation of `topMargin` and `leftMargin` assumes that the spider is initially centered within the wall. Ensure that this is the intended behavior, as any initial offset of the spider within the wall might affect the calculations.